Reworked the item weights so that we only need one list

ScipioWright opened the pull request:
Also removed some stuff that is no longer needed I think

ScipioWright committed:
Reworked the item weights so that we only need one list

heinermann commented on worlds/noita/Items.py:
```diff
@@ -40,12 +39,14 @@ def create_all_items(world: MultiWorld, player: int) -> None:
 
     # Create any non-progression repeat items (referred to as junk regardless of whether it's useful)
-    junk_pool = item_pool_weights[pool_option]
+    filler_pool = filler_weights
```
```suggestion
    filler_pool = filler_weights.copy()
```

Else it will modify the original which could cause unintended consequences.

ScipioWright committed:
Reworked adding orbs and the map perk in to use the names of the options rather than numbers

heinermann commented on worlds/noita/Items.py:
```diff
@@ -31,21 +31,25 @@ def create_all_items(world: MultiWorld, player: int) -> None:
 
     # Create any non-progression repeat items (referred to as junk regardless of whether it's useful)
-    junk_pool = item_pool_weights[pool_option]
+    filler_pool = filler_weights
+    if world.bad_effects[player].value == 0:
+        del filler_pool["Trap"]
```
Alternatively (maybe), remove "Trap" from the main weights list and replace this with

```py
filler_pool = filler_weights
if world.bad_effects[player].value:
    filler_pool = filler_weights | { "Trap": 15 }
```

ScipioWright committed:
Update worlds/noita/Items.py

Agree with suggestion, I figured what I was doing made a copy in the first place

Co-authored-by: Adam Heinermann <aheinerm@gmail.com>

ScipioWright committed:
Merge branch 'main' into sw_editing_things

ScipioWright committed:
Missed one of the options references that was tucked away in Rules.
Also made it abide by pep8 more since the ap guys like that
